usr/lib/common/mech_ec.c

```diff
@@ -1178,6 +1178,8 @@ CK_RV ecdh_pkcs_derive(STDLL_TokData_t *tokdata, SESSION *sess,
 
 end:
     free(derived_key);
+    if (temp_obj)
```

Comment:
We can not unconditionally free `temp_obj` here. When `object_mgr_create_final()` succeeds, then `temp_obj` must NOT be freed, because it has then be added to the btree and thus must stay available. 

I would suggest to free `temp_obj` and set `temp_obj` to NULL in lines 1153 and 1162 just like it is done in line 1172.  So just free it in the error case before going to `end`.

Reply:
You're right, I've improved the patch. Could you please have a look?
Thank you!

Reply:
It's done. Could you please have a look?
Thank you!